mix/QFunctionDefinition.cpp

```diff
@@ -21,19 +21,36 @@
 
 #include <libsolidity/AST.h>
 #include <libdevcrypto/SHA3.h>
+#include <libdevcore/Exceptions.h>
 #include "QVariableDeclaration.h"
 #include "QFunctionDefinition.h"
 
 using namespace dev::solidity;
 using namespace dev::mix;
 
-QFunctionDefinition::QFunctionDefinition(dev::solidity::FunctionDefinition const* _f, int _index): QBasicNodeDefinition(_f), m_index(_index), m_hash(dev::sha3(_f->getCanonicalSignature()))
+QFunctionDefinition::QFunctionDefinition(dev::solidity::FunctionDescription const& _f): QBasicNodeDefinition(_f.getDeclaration()), m_hash(dev::sha3(_f.getSignature()))
 {
-	std::vector<std::shared_ptr<VariableDeclaration>> parameters = _f->getParameterList().getParameters();
-	for (unsigned i = 0; i < parameters.size(); i++)
-		m_parameters.append(new QVariableDeclaration(parameters.at(i).get()));
+	FunctionDefinition const* funcDef;
```

Comment:
In the next iteration, this will hopefully use only FunctionType here, and we do not have to distinguish between the two.